extensions/amp-date-picker/1.0/dates-list.js

```diff
@@ -0,0 +1,149 @@
+import {addDays, isSameDay, isValid} from 'date-fns';
+import * as rrule from 'rrule';
+
+const rrulestr = rrule.default.rrulestr || rrule.rrulestr; // closure imports into .default, esbuild flattens a layer.
+
+/**
+ * RRULE returns dates as local time formatted at UTC, so the
+ * Date.prototype.getUTC* methods must be used to create a new date object.
+ * {@link https://github.com/jakubroztocil/rrule#important-use-utc-dates}
+ * @param {!Date} rruleDate
+ * @return {!Date}
+ */
+function normalizeRruleReturn(rruleDate) {
+  const year = rruleDate.getUTCFullYear();
+  const month = rruleDate.getUTCMonth();
+  const day = rruleDate.getUTCDate();
+  const hours = rruleDate.getUTCHours();
+  const minutes = rruleDate.getUTCMinutes();
+  const seconds = rruleDate.getUTCSeconds();
+  const ms = rruleDate.getUTCMilliseconds();
+  return new Date(year, month, day, hours, minutes, seconds, ms);
+}
+
+/**
+ * Tries to parse a string into an RRULE object.
+ * @param {string} str A string which represents a repeating date RRULE spec.
+ * @return {?RRule}
+ */
+function tryParseRrulestr(str) {
+  try {
+    return rrulestr(str, {});
+  } catch (e) {
+    return null;
+  }
+}
+
+/** @enum {string} */
+const DateType = {
+  INVALID: 'invalid',
+  RRULE: 'rrule',
+  DATE: 'date',
+};
+/**
+ * A class which wraps a list of moment dates or RRULE dates.
+ */
+export class DatesList {
```

Comment:
This is mostly copied from `0.1/dates-list.js`. It contains the same logic, but with `moment.js` stripped out.